render.js

```diff
@@ -0,0 +1,361 @@
+
+        // select todo_cards element
+        const todo_cards = document.querySelector('.todo-cards');
+        // console.log(todo_cards);
+
+        // create the unsaved card div element representing each card
+        const node = document.createElement('div');  
+        node.setAttribute('class', 'card-item unsaved-card'); 
+
+        // Start to fill in the unsaved card div with various elements 
+
+        // Create the input text area element and append it to the unsaved card div
+        const input_text = document.createElement('textarea');
+        input_text.setAttribute("class", "input-text");
+        input_text.placeholder = "What do you want to do?";
+        
+        // input_text.focus(); // NOT WORKING :(
+        // input_text.autofocus = true; // Autofocus processing was blocked because a document already has a focused element.
+                                    // Cannot even find the element that is focused
+
+        node.appendChild(input_text);
+
+        // Create a todo_buttons div containing the buttons in the unsaved cards div
+        const todo_buttons = document.createElement('div');
+        todo_buttons.setAttribute('class', 'todo-buttons');
+        
+        // Create the Add Task button and appending it to the todo_buttons
+        const add_task_button = document.createElement('button');
+        add_task_button.setAttribute('class', 'add-task-button');
+        add_task_button.appendChild(document.createTextNode("Add Task"));
+        todo_buttons.appendChild(add_task_button);
+
+        // followed by the event handler for the Add Task button 
+        add_task_button.addEventListener('click', () => {
+            add_task_button.disabled = true; // diasble the add task button so that multiple clicks don't create multiple cards
+            addTodo();
+            isCreateEnabled = true;
+        });
+        
+        // Create the delete button and appending it to the todo_buttons
+        const delete_button = document.createElement("input");
+        delete_button.setAttribute('class', 'delete-button');
+        delete_button.type = "image";
+        delete_button.name = "delete button";
+        delete_button.src="images/delete.svg";
+        delete_button.alt="delete button";
+        todo_buttons.appendChild(delete_button);
+
+        // followed by the event handler for the Delete button
+        delete_button.addEventListener("click", () => {
+            const item = document.querySelector('.unsaved-card');
+            item.remove();
+            isCreateEnabled = true;
+        });
+
+        // appending the todo_button div to the unsaved cards div
+        node.appendChild(todo_buttons);
+
+        // Prepend the element to the DOM as the first child of the 
+        // element referenced by the 'todo_cards' element
+        todo_cards.prepend(node);
+   }
+   
+   else
+   {
+       console.log("Create is disabled");
+   }
+
+}
+
+export async function renderAllData(search_text)
```

Comment:
If you want to pass a default argument for an empty string we can do so like this bhai.

Like this
`export const myFunction = (argument1 = "") => {...}`

Please rewrite all the functions in `function expression` format bhai to make your code more readable.

Like this

 ```suggestion
export const renderAllData = async (search_text = "") => {...}
```